rpc/namespaces/ethereum/eth/filters/filter_system.go

```diff
@@ -207,7 +208,32 @@ func (es *EventSystem) subscribeLogs(crit filters.FilterCriteria) (*Subscription
 		installed: make(chan struct{}, 1),
 		err:       make(chan error, 1),
 	}
-	return es.subscribe(sub)
+
+	return es.finishSubscribe(sub)
+}
+
+func (es *EventSystem) finishSubscribe(sub *Subscription) (*Subscription, pubsub.UnsubscribeFunc, error) {
```

Comment:
Nit: I think `finishSubscribe` is a bit misleading name comparing to what this function does. WDYT about `subscribeWithCleanup`?

Reply:
sure thing, I thought of it finalizing the subcription, but happy to rename into subscribeWithCleanup, let me change it.